src/main/kotlin/org/rust/ide/inspections/RsVariableMutableInspection.kt

```diff
@@ -0,0 +1,58 @@
+package org.rust.ide.inspections
+
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiRecursiveElementVisitor
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.ide.annotator.fixes.RemoveMutableFix
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.isMut
+import org.rust.lang.core.psi.ext.parentOfType
+import java.util.*
+
+private fun findBlock(expr: PsiElement) = PsiTreeUtil.getNonStrictParentOfType(expr, RsBlock::class.java)
+
+class RsVariableMutableInspection : RsLocalInspectionTool() {
+    override fun getDisplayName() = "No mutable required"
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean) =
+        object : RsVisitor() {
+            override fun visitPatBinding(o: RsPatBinding) {
+                if (!o.isMut) return
+                val occurrences = findOccurrences(o.identifier)
+                var isMutableNeeded = false
+                if (!occurrences.isEmpty()) {
+                    for (occurrence in occurrences) {
+                        val parent = occurrence.parentOfType<RsBinaryExpr>() ?: continue
+                        if (parent.left == occurrence) {
+                            isMutableNeeded = true
+                            break
+                        }
+                    }
+                }
+                if (!isMutableNeeded) {
+                    holder.registerProblem(o, "No mutable required for ${o.identifier.text}", RemoveMutableFix(o))
```

Comment:
Hm, I think this'll give false positives for stuff like

```Rust
struct S; 
impl S {  fn needs_mut(&mut self) {} }

fn main() {
  let s = S;
  s.needs_mut()
}
```


I am not actually sure that we can implement this inspection without false positives reliably at the moment...